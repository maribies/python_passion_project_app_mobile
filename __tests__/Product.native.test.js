import React from "react";
import { create } from "react-test-renderer";
import { Product } from "../Components/Product.native";
import products from "../Fixtures/products.json";

const product = products[0];

test('renders correctly with no data', () => {
  const tree = create(<Product />).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({Text:"Opps! No Data!"})
});

test('renders correctly with undefined data', () => {
  const tree = create(<Product product={undefined}/>).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({Text:"Opps! No Data!"})
});

test('renders correctly with data', () => {
  const tree = create(<Product product={product} />).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({Text:"Chanel"})
});

// TODO: Should be testing this method directly but keep getting mock function errors.
test('getStockDetails mock returns details', () => {
  const stock = [{"color":"black", "quantity": 3}, {"color":"white", "quantity": 4}];

  const testGetStockDetails = jest.fn(stock => stock.map(data => {
    const { color, quantity } = data;

    return `${color} ${quantity}`
  }));

  testGetStockDetails(stock);

  expect(testGetStockDetails).toHaveReturnedWith(["black 3", "white 4"])
});
