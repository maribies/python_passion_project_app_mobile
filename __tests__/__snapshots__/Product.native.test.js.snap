// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Helvetica",
      "fontSize": 16,
      "fontWeight": "normal",
      "lineHeight": 21,
    }
  }
>
  Opps! No Data!
</Text>
`;

exports[`renders correctly with no data 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Helvetica",
      "fontSize": 16,
      "fontWeight": "normal",
      "lineHeight": 21,
    }
  }
>
  Opps! No Data!
</Text>
`;

exports[`renders correctly with undefined data 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Helvetica",
      "fontSize": 16,
      "fontWeight": "normal",
      "lineHeight": 21,
    }
  }
>
  Opps! No Data!
</Text>
`;
